src/test/java/com/jpop/productservice/integration/ProductServiceApplicationIT.java

```diff
@@ -23,18 +32,54 @@
     @Autowired
     TestRestTemplate testRestTemplate;
 
+    @Autowired
+    ProductDetailService productDetailService;
+
+    @TestConfiguration
+    static class ProductServiceApplicationITConfig {
+
+        @Bean
+        public ProductDetailService productDetailService() {
+            return new ProductDetailServiceImpl();
+        }
+    }
+
     @Test
     public void testGetProducts() {
         HttpHeaders headers = new HttpHeaders();
         headers.setContentType(MediaType.APPLICATION_JSON);
         headers.set("Accept", MediaType.APPLICATION_JSON_VALUE);
         HttpEntity httpEntity = new HttpEntity(null, headers);
 
-        ResponseEntity<String> responseEntity = testRestTemplate.exchange(createUrl(), HttpMethod.GET, httpEntity, String.class);
+        ResponseEntity<String> responseEntity = testRestTemplate.exchange(createUrl("/api/v1/products/"), HttpMethod.GET, httpEntity, String.class);
         assertEquals("Status code is invalid", 200, responseEntity.getStatusCode().value());
     }
 
-    private String createUrl() {
-        return "http://localhost:" + port + "/api/v1/products/";
+    @Test
+    public void testGetProductByID() {
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        headers.set("Accept", MediaType.APPLICATION_JSON_VALUE);
+        HttpEntity httpEntity = new HttpEntity(null, headers);
+
+        ResponseEntity<String> responseEntity = testRestTemplate.exchange(createUrl("/api/v1/products/1"), HttpMethod.GET, httpEntity, String.class);
+        assertEquals("Status code is invalid", 200, responseEntity.getStatusCode().value());
+        assertNotNull("Data is null for the product ID", responseEntity.getBody());
+    }
+
+    @Test
+    public void testInsertProduct() {
+        Product product = new Product(3, "Shirt", "Adidas", new BigDecimal(123.98));
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        headers.set("Accept", MediaType.APPLICATION_JSON_VALUE);
+        HttpEntity httpEntity = new HttpEntity(new Gson().toJson(product), headers);
+        ResponseEntity<Product> responseEntity = testRestTemplate.exchange(createUrl("/api/v1/products/"), HttpMethod.POST, httpEntity, Product.class);
+        assertEquals("Status code is invalid", 201, responseEntity.getStatusCode().value());
+        assertEquals("Name is not correct",product.getName(), responseEntity.getBody().getName());
+    }
+
+    private String createUrl(String path) {
+        return "http://localhost:" + port + path;
     }
 }
```

Comment:
Are these tests using MySQL? Using mysql will persist the data from tests which we don't want to happen. Please use in memory database for tests (with separate db property files in `src/main/resources` and  `src/test/resources`). Once you have done that you can create/update records from above test cases) and do meaningful assertion. 

Again, this can be a future task.

Reply:
I am using embedded server only, but I will another file in test folder only

Reply:
You should have used MYSQL. Liquibase doesn't make sense with embedded DB. Please add it as task in week3.